shared/datastore.go

```diff
@@ -42,6 +42,7 @@ type Datastore interface {
 	Done() interface{}
 	NewQuery(typeName string) Query
 	NewKey(typeName string, id int64) Key
+	ReserveKey(typeName string) (Key, error)
```

Comment:
Nit: name this `ReserveID` as it can only reserve a numeric ID automatically.

Reply:
Done.